extensions/notification/NotificationQueuePopulator.js

```diff
@@ -217,91 +217,88 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
+                    eventType,
+                    versionId,
+                    dateTime,
+                };
+                this.log.debug('validating entry', {
+                    method: 'NotificationQueuePopulator._processObjectEntry',
+                    bucket,
+                    key,
+                    eventType,
+                });
+                const pushedToTopic = new Map();
+                this.notificationConfig.destinations.forEach(destination => {
+                    const topic = destination.internalTopic ||
+                        this.notificationConfig.topic;
+                    if (pushedToTopic[topic]) {
+                        return undefined;
+                    }
+                    const queueConfig = config.notificationConfiguration.queueConfig.filter(
+                        c => c.queueArn.split(':').pop() === destination.resource
+                    );
+                    if (!queueConfig.length) {
+                        return undefined;
+                    }
+                    const destConfig = {
+                        bucket,
+                        notificationConfiguration: {
+                            queueConfig,
+                        },
+                    };
+                    const { isValid, matchingConfig } = configUtil.validateEntry(destConfig, ent);
+                    if (isValid) {
+                        const message
+                            = messageUtil.addLogAttributes(value, ent);
+                        this.log.info('publishing message', {
+                            method: 'NotificationQueuePopulator._processObjectEntry',
+                            bucket,
+                            key: message.key,
+                            versionId,
+                            eventType,
+                            eventTime: message.dateTime,
+                            matchingConfig,
+                        });
+                        this.publish(topic,
+                            `${bucket}/${message.key}`,
+                            JSON.stringify(message));
+                        pushedToTopic[topic] = true;
+                    }
+                    return undefined;
+                });
+            }
             return undefined;
-        }
-        const baseKey = this._extractVersionedBaseKey(key);
-        const versionId = this._getVersionId(value, overheadFields);
-        const dateTime = this._getEventDateTime(value, overheadFields);
-        const config = await this.bnConfigManager.getConfig(bucket);
-        if (config && Object.keys(config).length > 0) {
-            const ent = {
-                bucket,
-                key: baseKey,
-                eventType,
-                versionId,
-                dateTime,
-            };
-            this.log.debug('validating entry', {
+        } catch (error) {
+            const errorMessage = error?.toString?.() || JSON.stringify(error) || 'Unknown error';
```

Comment:
This looks hacky, don't we have better ways to handle it? If that's an error it should always have a `.message`, no?